lib/core/src/receive_swap.rs

```diff
@@ -260,12 +260,116 @@ impl ReceiveSwapStateHandler {
                 err: format!("Cannot transition from {from_state:?} to Complete state"),
             }),
 
-            (Created, TimedOut) => Ok(()),
+            (Created | TimedOut, TimedOut) => Ok(()),
             (_, TimedOut) => Err(PaymentError::Generic {
                 err: format!("Cannot transition from {from_state:?} to TimedOut state"),
             }),
 
+            (Complete, Failed) => Err(PaymentError::Generic {
+                err: format!("Cannot transition from {from_state:?} to Failed state"),
+            }),
             (_, Failed) => Ok(()),
         }
     }
 }
+
+pub(crate) mod test_utils {
```

Comment:
As far as I see the test_utils contains mocks for the various services.
What do you think about centralizing these on one file?

Reply:
Agreed :+1: